assets/js/components/Dashboard/PaMessaging/NePaMessagePage/NewPaMessagePage.tsx

```diff
@@ -0,0 +1,145 @@
+      dayLabel: string;
+    }
+  | {
+      type: "SET_SELECTED_DAYS";
+      days: number[];
+    }
+  | {
+      type: "SET_PRIORITY";
+      priority: number;
+    }
+  | {
+      type: "SET_INTERVAL";
+      interval: number;
+    }
+  | {
+      type: "SET_VISUAL_TEXT";
+      visualText: string;
+    }
+  | {
+      type: "SET_PHONETIC_TEXT";
+      phoneticText: string;
+    };
+
+const reducer = (
+  state: NewPaMessagePageState,
+  action: NewPaMessagePageReducerAction,
+) => {
+  switch (action.type) {
+    case "SET_START_DATE":
+      return fp.set("selectedStartDate", action.date, state);
+    case "SET_START_TIME":
+      return fp.set("selectedStartTime", action.time, state);
+    case "SET_END_DATE":
+      return fp.set("selectedEndDate", action.date, state);
+    case "SET_END_TIME":
+      return fp.set("selectedEndTime", action.time, state);
+    case "SET_DAY_LABEL":
+      return fp.set("selectedDayLabel", action.dayLabel, state);
+    case "SET_SELECTED_DAYS":
+      return fp.set("selectedDays", action.days, state);
+    case "SET_PRIORITY":
+      return fp.set("selectedPriority", action.priority, state);
+    case "SET_INTERVAL":
+      return fp.set("selectedInterval", action.interval, state);
+    case "SET_VISUAL_TEXT":
+      return fp.set("visualText", action.visualText, state);
+    case "SET_PHONETIC_TEXT":
+      return fp.set("phoneticText", action.phoneticText, state);
+  }
+};
+
+const NewPaMessagePage: ComponentType = () => {
+  const now = moment();
+  const initialState: NewPaMessagePageState = {
+    selectedStartDate: now.format("L"),
+    selectedStartTime: now.format("HH:mm"),
+    selectedEndDate: now.format("L"),
+    selectedEndTime: now.add(1, "hour").format("HH:mm"),
+    selectedDayLabel: "All days",
+    selectedDays: [1, 2, 3, 4, 5, 6, 7],
+    selectedPriority: 2,
+    selectedInterval: 4,
+    visualText: "",
+    phoneticText: "",
+  };
+
+  const [state, dispath] = useReducer(reducer, initialState);
+  const navigate = useNavigate();
+
+  return (
+    <div className="new-pa-message-page">
+      <div className="new-pa-message-page__header">New PA/ESS message</div>
+      <Container fluid>
+        <Row md="auto" className="align-items-center">
+          <Button variant="link" className="pr-0">
+            Associate with alert
+          </Button>
+          (Optional)
+        </Row>
+        <Row>
+          <div className="new-pa-message-page__associate-alert-subtext">
+            Linking will allow you to share end time with alert, and import
+            location and message.
+          </div>
+        </Row>
+        <WhenCard pageState={state} dispatch={dispath} />
+        <WhereCard pageState={state} dispatch={dispath} />
+        <MessageCard pageState={state} dispatch={dispath} />
+        <Row
+          md="auto"
+          className="justify-content-end new-pa-message-page__form-buttons"
+        >
+          <Button className="cancel-button" onClick={() => navigate(-1)}>
```

Comment:
**question:** should this link to a specific page? what does this button do if someone opens a link to this page from an external program like Slack or Teams?

**suggestion:** if changing to link to a specific page i'm a big fan of using links/anchor tags to perform navigation

Reply:
I hadn't thought about what this would do if this page was opened in a new tab. I'll adjust this to navigate to `/pa-messages` explicitly in case there is no navigation history to go back to.

Reply:
Actually, I'll confirm with Mary what she wants here. I also see value in closing the tab if there is no history so my original idea may not be what she wants.